src/main.ts

```diff
@@ -121,6 +121,32 @@ export async function activate(context: vscode.ExtensionContext): Promise<CSharp
         eventStream.post(new ActivationFailure());
     }
 
+    if (!isSupportedPlatform(platformInfo)) {
+        const platform: string = platformInfo.platform ? platformInfo.platform : "this platform";
+        const architecture: string = platformInfo.architecture ? platformInfo.architecture : " and architecture";
```

Comment:
Might want to `"<Unknown processor architecture>"` rather then `"and architecture"` since, if the platform is understood, 'and architecture would be a bit weird -- "... is incompatible on Windows and architecture"